### DISCUSSION TITLE
Custom paragraph node using node replacement system does not show placeholder text

### DISCUSSION THREAD

**Discussion by IsmailElShinnawy** [facebook/lexical#3619] [Q&A] [answered]
Hi,

I have a custom node for a paragraph `MyParagraphNode` that extends lexical's `ParagraphNode`, and I use the node replacement system to replace lexical's `ParagraphNode` with my custom node, but that causes the placeholder - that is rendered when the editor is "empty" - to not show up. This is because internally `$canShowPlaceholder` uses `$isParagraphNode`. Is there a current solution for this?

Thank you in advance,

**Comment by moy2010** [comment, marked as answer]
Having `$canShowPlaceholder` to use `isParagraphNode` is actually [the fix for this issue](https://github.com/facebook/lexical/pull/3557). If your custom node `MyParagraphNode` is extending lexical's `ParagraphNode` as you say, then it is an instance of it.

**Reply by IsmailElShinnawy** [reply]
Oh I see, when I upgraded to 0.7.3 it worked. Thank you so much for your reply :)
